llvm/lib/Transforms/InstCombine/InstCombineAddSub.cpp

```diff
@@ -2631,6 +2631,21 @@ Instruction *InstCombinerImpl::visitSub(BinaryOperator &I) {
                                                /* IsNUW */ false))
       return replaceInstUsesWith(I, Res);
 
+  if (match(Op0, m_ZExt(m_PtrToInt(m_Value(LHSOp)))) &&
+      match(Op1, m_ZExtOrSelf(m_PtrToInt(m_Value(RHSOp))))) {
```

Comment:
Not sure if we need one-use check here.

Reply:
What do you have in mind ? It looks unnecessary as neither Op0 nor Op1 is replaced. The pattern-matching replaces only subtract with an offset

Reply:
We don't have a one-use check for the generic cases above, so I'd say "no" for now at least.

Reply:
> The pattern-matching replaces only subtract with an offset

`EmitGEPOffset` generates additional instructions.